### PR TITLE
ARI handling updates for containers in text form

### PR DESCRIPTION
**Pull request by justinethier**
This MR adds addresses ARI handling of AC, AM, and TBL objects:

- Updated parsing of literal parameter lists to support case-insensitive keys and to raise an error if a duplicate key is found. This change is applicable to TBL objects as well as other objects that use parameter lists. 
- Added unit tests for the above.
- Added dedicated decoding test cases for all three object types.
- Added additional unit tests to handle edge cases with AC and TBL objects.

In addition, this PR adds the ability to handle object type names in a case insensitive manner and provides tests for them.

### CODE REVIEW

File: src/cace/ari/text_util.h [hunk 1]
```diff
@@ -140,6 +140,22 @@ int slash_unescape(string_t out, const cace_data_t *in);
  */
 void strip_space(string_t out, const char *in, size_t in_len);
 
+/** Convert a text string to lowercase.
+ * This is written to work on byte strings, not unicode.
+ *
+ * @param[out] out The output buffer, which will be replaced.
+ * @return Zero upon success.
+ */
+int cace_string_tolower(string_t out);
```
Comment (BrianSipos) [inline src/cace/ari/text_util.h:149]: If it really cannot fail then these can return void and use precond macro `CHKVOID()`. Since the return values were not being checked anyway.

File: src/cace/ari/text_util.h [hunk 2]
```diff
@@ -140,6 +140,22 @@ int slash_unescape(string_t out, const cace_data_t *in);
...
+
+/** Convert a text string to uppercase.
+ * This is written to work on byte strings, not unicode.
+ *
+ * @param[out] out The output buffer, which will be replaced.
+ * @return Zero upon success.
+ */
+int cace_string_toupper(string_t out);
```
Comment (BrianSipos) [inline src/cace/ari/text_util.h:157]: Same as other function.

**Review by BrianSipos — CHANGES_REQUESTED** [review]
Other than the new return types, this looks good

File: src/cace/ari/text_util.h (hunk 1 above)
Comment (justinethier) [reply src/cace/ari/text_util.h]: Thanks @BrianSipos, changed both functions to `void`.

File: src/cace/ari/text_util.h (hunk 2 above)
Comment (justinethier) [reply src/cace/ari/text_util.h]: Updated this one as well, thanks.

**Review by BrianSipos — APPROVED** [review]
